Algorithm.CSharp/OptionIndicatorsRegressionAlgorithm.cs

```diff
@@ -42,11 +46,12 @@ public override void Initialize()
             _option = QuantConnect.Symbol.CreateOption("AAPL", Market.USA, OptionStyle.American, OptionRight.Put, 505m, new DateTime(2014, 6, 27));
             AddOptionContract(_option);
 
-            var interestRateProvider = new InterestRateProvider();
-            var dividendYieldProvider = new DividendYieldProvider(_aapl);
-
-            _impliedVolatility = new ImpliedVolatility(_option, interestRateProvider, dividendYieldProvider, OptionPricingModelType.BlackScholes, 2);
-            _delta = new Delta(_option, interestRateProvider, dividendYieldProvider, OptionPricingModelType.BinomialCoxRossRubinstein, OptionPricingModelType.BlackScholes);
+            _impliedVolatility = IV(_option, period: 2);
+            _delta = D(_option, optionModel: OptionPricingModelType.BinomialCoxRossRubinstein, ivModel: OptionPricingModelType.BlackScholes);
+            _gamma = G(_option, optionModel: OptionPricingModelType.ForwardTree, ivModel: OptionPricingModelType.BlackScholes);
+            _vega = V(_option, optionModel: OptionPricingModelType.ForwardTree, ivModel: OptionPricingModelType.BlackScholes);
+            _theta = T(_option, optionModel: OptionPricingModelType.ForwardTree, ivModel: OptionPricingModelType.BlackScholes);
+            _rho = R(_option, optionModel: OptionPricingModelType.ForwardTree, ivModel: OptionPricingModelType.BlackScholes);
```

Comment:
Cool! I think it would be awesome to add a similar regression algorithm with the mirror option contract too, as a test case and example both work happily 👍